include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -56,18 +57,74 @@ struct GemmKernel
                                                             index_t K,
                                                             index_t stride_A,
                                                             index_t stride_B,
-                                                            index_t stride_C)
+                                                            index_t stride_C,
+                                                            index_t KBatch)
     {
-        return GemmCommonKargs{a_ptr, b_ptr, c_ptr, M, N, K, stride_A, stride_B, stride_C};
+        return GemmCommonKargs{a_ptr, b_ptr, c_ptr, M, N, K, stride_A, stride_B, stride_C, KBatch};
     }
 
     CK_TILE_HOST_DEVICE static constexpr index_t GetSmemSize()
     {
         return max(GemmPipeline::GetSmemSize(), EpiloguePipeline::GetSmemSize());
     }
 
+    struct SplitKBatchOffset
+    {
+        __device__ SplitKBatchOffset(GemmCommonKargs& kargs)
+        {
+            constexpr auto K1   = TilePartitioner::BlockGemmShape::WarpTile::at(number<2>{});
+            const index_t K_t   = kargs.KBatch * K1;
+            const index_t KRead = (kargs.K + K_t - 1) / K_t * K1;
+
+            if constexpr(std::is_same_v<tensor_layout::gemm::RowMajor, ALayout>)
+            {
+                a_k_split_offset = blockIdx.z * KRead;
+            }
+            else if constexpr(std::is_same_v<tensor_layout::gemm::ColumnMajor, ALayout>)
+            {
+                a_k_split_offset = blockIdx.z * KRead * kargs.stride_A;
+            }
+
+            if constexpr(std::is_same_v<tensor_layout::gemm::RowMajor, BLayout>)
+            {
+                b_k_split_offset = blockIdx.z * KRead * kargs.stride_B;
+            }
+            else if constexpr(std::is_same_v<tensor_layout::gemm::ColumnMajor, BLayout>)
+            {
+                b_k_split_offset = blockIdx.z * KRead;
+            }
+
+            if(blockIdx.z < static_cast<uint32_t>(kargs.KBatch - 1))
+            {
+                SplittedK = KRead;
+            }
+            else
+            {
+                SplittedK = kargs.K - KRead * (kargs.KBatch - 1);
+            }
+        }
+
+        index_t a_k_split_offset;
+        index_t b_k_split_offset;
+        index_t SplittedK;
+    };
+
     CK_TILE_HOST static bool IsSupportedArgument(const GemmCommonKargs& kargs)
     {
+        constexpr bool is_output_c_reg_transposed =
+            EpiloguePipeline::IsOutputTransposed() != GemmPipeline::IsTransposeC();
+        if constexpr(!((GemmPipeline::VectorSizeC % 2 == 0 &&
+                        std::is_same_v<CLayout, tensor_layout::gemm::RowMajor> &&
+                        is_output_c_reg_transposed) ||
+                       !(std::is_same_v<CDataType, half_t> ||
+                         std::is_same_v<CDataType, bfloat16_t>)))
```

Comment:
You may use `bf16_t` alias: https://github.com/ROCm/composable_kernel/blob/6f677a8ab9ba0cb1606ff32aa55bdcb9ca58c954/include/ck_tile/core/numeric/bfloat16.hpp#L106